### PR TITLE
feat(#1398): Add keyPress event for input

### PR DESCRIPTION
**Pull request by amar-ABgov**

### CODE REVIEW

**Comment by Radha-Alberta** [comment]
- inputbox value and value in event triggered are different, Is this expected ?

image

File: libs/react-components/src/lib/input/input.tsx [hunk 1]
```diff
@@ -104,11 +105,13 @@ export interface InputProps extends BaseProps {
   step?: number;
   onFocus?: OnFocus;
   onBlur?: OnBlur;
+  onKeyPress?: OnKeyPress
```
Comment (ArakTaiRoth) [inline libs/react-components/src/lib/input/input.tsx:108]: Question, I must have missed this earlier. Why don't we move `onChange`, `onFocus`, `onBlur`, and now `onKeyPress` to `BaseProps`, so we don't have to repeatedly add it to every other Input type we have?

Comment (chrisolsen) [reply libs/react-components/src/lib/input/input.tsx:108]: For consistency it would make sense to also have the onKeyPress o on the TextArea as well.

**Review by ArakTaiRoth — DISMISSED** [review]

File: libs/react-components/src/lib/input/input.tsx [hunk 2]
```diff
@@ -194,17 +189,24 @@ export function GoAInput({
       onBlur?.(name, value);
     };
 
+    const keypressListner = (e: unknown) => {
```
Comment (chrisolsen) [inline libs/react-components/src/lib/input/input.tsx:192]: Small spelling mistake `Listner`

File: libs/react-components/src/lib/textarea/textarea.tsx [hunk 3]
```diff
@@ -70,9 +72,16 @@ export function GoATextArea({
       onChange(name, value);
     };
 
+    const keypressListner = (e: unknown) => {
```
Comment (chrisolsen) [inline libs/react-components/src/lib/textarea/textarea.tsx:75]: Same spelling mistake

**Pull request merged**